agent/apphealth.go

```diff
@@ -0,0 +1,169 @@
+package agent
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/retry"
+)
+
+type WorkspaceAppHealthReporter func(ctx context.Context)
+
+func NewWorkspaceAppHealthReporter(logger slog.Logger, client *codersdk.Client) WorkspaceAppHealthReporter {
```

Comment:
If this is just initialized inside `cli/agent.go`, could we just use it inside an agent instead? Doesn't seem like it needs to be exported.

Reply:
I also see this doesn't have any tests, which seems sketchy.

Reply:
Added some cool tests for this